tour_model_eval/load_predict.py

```diff
@@ -84,9 +111,36 @@ def predict_labels(trip):
 if __name__ == '__main__':
     participant_uuid_obj = list(edb.get_profile_db().find({"install_group": "participant"}, {"user_id": 1, "_id": 0}))
     all_users = [u["user_id"] for u in participant_uuid_obj]
+
+    # case 1: the new trip matches a bin from the 1st round and a cluster from the 2nd round
     user = all_users[0]
     radius = 100
     trips = preprocess.read_data(user)
     filter_trips = preprocess.filter_data(trips, radius)
     new_trip = [filter_trips[4]]
+    # result is [{'labels': {'mode_confirm': 'shared_ride', 'purpose_confirm': 'church', 'replaced_mode': 'drove_alone'},
+    # 'p': 0.9333333333333333}, {'labels': {'mode_confirm': 'shared_ride', 'purpose_confirm': 'entertainment',
+    # 'replaced_mode': 'drove_alone'}, 'p': 0.06666666666666667}]
+    print(predict_labels(new_trip))
+
+    # case 2: no existing files for the user who has the new trip:
+    # 1. the user is invalid(< 10 existing fully labeled trips, or < 50% of trips that fully labeled)
+    # 2. the user doesn't have common trips
+    user = all_users[1]
+    trips = preprocess.read_data(user)
+    new_trip = [trips[0]]
+    # result is []
     print(predict_labels(new_trip))
+
+    # case3: the new trip is novel trip(doesn't fall in any 1st round bins)
+    user = all_users[0]
+    radius = 100
+    trips = preprocess.read_data(user)
+    filter_trips = preprocess.filter_data(trips, radius)
+    new_trip = [filter_trips[0]]
+    # result is []
+    print(predict_labels(new_trip))
+
+    # case 4: the new trip falls in a 1st round bin, but predict to be a new cluster in the 2nd round
+    # result is []
+    # no example for now
```

Comment:
Glad to see some testing done, even if it is not fully complete and is not in the unit test framework.